include/zephyr/rtio/rtio.h

```diff
@@ -38,6 +38,8 @@
 #include <zephyr/sys/iterable_sections.h>
 #include <zephyr/sys/mpsc_lockfree.h>
 
+#include <../kernel/include/timeout_q.h>
```

Comment:
I'm game for this, but its going to need some scrutiny and/or making it less private and/or moving rtio into the kernel part of the tree perhaps (though this will bring yet more scrutiny to this code).

I like the concept though, as noted timeouts already *do* the shortest time queuing and are well tested. So makes good sense to reuse as @bjarki-andreasen noted, the concern before was the size cost.